src/lib.rs

```diff
@@ -473,10 +473,17 @@ impl<T: Counter> Histogram<T> {
                 .map_err(|_| AdditionError::ResizeFailedUsizeTypeTooSmall)?;
         }
 
-        if self.bucket_count == source.bucket_count
+        let matching_buckets = self.bucket_count == source.bucket_count
             && self.sub_bucket_count == source.sub_bucket_count
-            && self.unit_magnitude == source.unit_magnitude
-        {
+            && self.unit_magnitude == source.unit_magnitude;
+        if matching_buckets && self.is_empty() {
```

Comment:
I'm confused — if `is_empty()` returns `true`, then surely that means we can just set all the fields to `0`, no? In which case we don't even need `copy_from_slice`, we can just `memset` all of it to 0?

Reply:
self is the "target" histogram, so by checking if it's all 0 we can add the "source" in the form of a mem copy. I suppose at the top of this function we could check if the target histogram is empty and do nothing as well.

Reply:
Ah, yes, had my references backwards. I think adding a check for if the target is empty to optimize that path makes sense :+1:

Reply:
Done